### PR TITLE
TST: GH50347 create test for groupby sum of boolean values.

### LINKED ISSUE #50347: BUG: Groupby then sum boolean column gives too large a sum
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [x] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

n = 32

df = pd.DataFrame({'label': [1]*n,'bool_column': [True]*(n)})

print(df.assign(bool_column_dupe=(df.bool_column==True)).groupby("label").bool_column_dupe.sum())
```

### Issue Description

Summing a boolean column after a groupby gives an unreasonably large and incorrect sum.
This is the result from the above code:
`
1    8128
Name: bool_column_dupe, dtype: int64
`

The issue does not occur if I instead add an `astype(int)`:
`print(df.assign(bool_column_dupe(df.bool_column==True).astype(int)).groupby("label").bool_column_dupe.sum())`

The issue does not occur if I don't use the `.assign`:
`print(df.groupby("label").bool_column.sum())`

The issue does not occur with `n<32`.

The issue only occurs with numpy v1.24.0.
It does not occur with numpy v1.23.5.

### Expected Behavior

Expected to see:
`1    32
Name: bool_column_dupe, dtype: int64`

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7
python           : 3.9.13.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.1.0
Version          : Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:54 PDT 2022; root:xnu-8792.41.9~2/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_GB.UTF-8
pandas           : 1.5.2
numpy            : 1.24.0
pytz             : 2022.7
dateutil         : 2.8.2
setuptools       : 60.2.0
pip              : 21.3.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by Daquisu**
- [x] closes #50347
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Does not apply:
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -2828,3 +2828,11 @@ def test_groupby_index_name_in_index_content(val_in, index, val_out):
...
     expected = expected.to_frame()
     tm.assert_frame_equal(result, expected)
+
+
+def test_sum_of_booleans():
+    # GH 50347
+    n = 32
+    df = DataFrame({"groupby_col": 1, "bool": [True] * n})
+    df["bool"] = df["bool"].eq(True)
+    assert df.groupby("groupby_col").sum().iloc[0, 0] == 32
```
Comment (phofl) [inline pandas/tests/groupby/test_groupby.py:2838]: We would want to compare the whole object

Comment (Daquisu) [reply pandas/tests/groupby/test_groupby.py]: Of course. Done.

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -2828,3 +2828,13 @@ def test_groupby_index_name_in_index_content(val_in, index, val_out):
     result = series.to_frame().groupby("blah").sum()
     expected = expected.to_frame()
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("n", [1, 10, 100, 1000])
```
Comment (agaventa) [inline pandas/tests/groupby/test_groupby.py:2833]: would it be worth putting in n=32 as this was the original value which triggered the bug?
https://github.com/pandas-dev/pandas/issues/50347

Comment (Daquisu) [reply pandas/tests/groupby/test_groupby.py]: For what it's worth it happened for n ≥ 32, but indeed it is relevant to add this case. Done.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @Daquisu

**Pull request merged**